Remove RealtimeClient endpoint parameter name underscore prefix from initializers

foodisbeast opened the pull request:
## What kind of change does this PR introduce?

Removes the `_` prefix from the RealtimeClient initializers

## What is the current behavior?
RealtimeClient initializer has parameter with underscore prefix
```swift
init(_ endpoint: String, ...)
```

[SupabaseClient](https://github.com/supabase-community/supabase-swift/blob/master/Sources/Supabase/SupabaseClient.swift) initializer in `main` branch of supabase-swift uses `endpoint` without `_` prefix.
```swift
/// Realtime client for Supabase
  public var realtime: RealtimeClient {
    RealtimeClient(
      endPoint: realtimeURL.absoluteString,
      params: defaultHeaders
    )
  }
```


## What is the new behavior?

Remove `_` prefix from RealtimeClient Initializers.
```swift
init(endpoint: String, ...)
```

## Additional context

This fix for #21 should enable usage in current version (0.3.0) of `supabase-swift`

foodisbeast committed:
remove endpoint parameter name underscore prefix

grsouza commented:
Let's keep the initializer and add a new one without the `_` and mark it as deprecated.

foodisbeast committed:
Revert "remove endpoint parameter name underscore prefix"

This reverts commit 478c72e58908c05ba61d00f6e77b22c394fb1950.

foodisbeast committed:
add deprecated initializer to RealtimeClient for compatibility

foodisbeast commented:
Did as you suggested @grsouza! Reverted previous commit adding removing `_` prefix, and added deprecated initializer.

foodisbeast committed:
Revert "add deprecated initializer to RealtimeClient for compatibility"

This reverts commit 606115265190e4c10c73ee81dce2c08dd7ad839f.

foodisbeast committed:
add deprecated initializer to RealtimeClient for compatibility with supabase-swift 0.3.0

foodisbeast commented:
Did as you suggested @grsouza! Reverted previous commit removing `_` prefix, and added deprecated initializer matching current usage in `supabase-swift` v0.3.0 .
